R/plot_section.R

```diff
@@ -0,0 +1,154 @@
+#' # Sample tetrapod occurrences
+#' tetrapod_names <- tetrapods$accepted_name[1:50]
+#' # Simulate bed numbers
+#' beds_sampled <- sample.int(n = 10, size = 50, replace = TRUE)
+#' # Simulate certainty values
+#' certainty_sampled <- sample(x = 0:1, size = 50, replace = TRUE)
+#' #Combine into data frame
+#' occdf <- data.frame(taxon = tetrapod_names, bed = beds_sampled,
+#' certainty = certainty_sampled)
+#' # Plot stratigraphic ranges
+#' tax_range_strat(occdf)
+#' tax_range_strat(occdf, certainty = "certainty")
+#' tax_range_strat(occdf, certainty = "certainty", by = "LAD")
+#' tax_range_strat(occdf, certainty = "certainty", by = "name")
+#'
+#' @export
+tax_range_strat <- function (occdf, name = "taxon", level = "bed",
+                             certainty = FALSE, by = "FAD",
+                             units = "height", label = "Section A")
+{
+  if (is.data.frame(occdf) == FALSE) {
+    stop("`occdf` should be a dataframe")
+  }
+
+  if (!is.numeric(occdf[, level])) {
+    stop("`level` must be of class numeric.")
+  }
+
+  if (any(c(name, level) %in% colnames(occdf) == FALSE)) {
+    stop("Either `name` or `level` is not a named column in\n`occdf`")
+  }
+
+  if (certainty != FALSE && certainty %in% colnames(occdf) == FALSE) {
+    stop("`certainty` is not a named column in\n`occdf`")
+  }
+
+  if (any(is.na(occdf[, name]))) {
+    stop("The `name` column contains NA values")
+  }
+
+  if (any(is.na(occdf[, level]))) {
+    stop("The `level` column contains NA values")
+  }
+
+  if (!by %in% c("name", "FAD", "LAD")) {
+    stop("`by` must be either \"FAD\", \"LAD\", or \"name\"")
+  }
+
+  if (!units %in% c("height", "beds")) {
+    stop("`units` must be either \"height\" or \"beds\"")
+  }
+
+  #List and order unique taxa
+  unique_taxa <- unique(occdf[, name])
+  unique_taxa <- unique_taxa[order(unique_taxa)]
+
+  #Create object to hold information
+  ranges <- data.frame(taxon = unique_taxa, min_bin = NA, max_bin = NA)
+
+  #Populate nested list
+  for (i in seq_along(unique_taxa)) {
+    occ_filter <- occdf[(occdf[, name] == unique_taxa[i]),]
+    ranges[i,2] <- min(occ_filter[level])
+    ranges[i,3] <- max(occ_filter[level])
+  }
+
+  #Reorder lists
+  if (by == "LAD") {
+    ranges <- ranges[order(ranges$min_bin), ]
+    ranges <- ranges[order(ranges$max_bin), ]
+  }
+
+  if (by == "FAD") {
+    ranges <- ranges[order(ranges$max_bin), ]
+    ranges <- ranges[order(ranges$min_bin), ]
+  }
+
+  #Add ID numbers
+  ranges$ID <- c(1:length(unique_taxa))
+  labels <- ranges[, c(1,4)]
+  occdf <- merge(occdf, labels, by.x = name, by.y = "taxon")
+
+  #Split certain and uncertain occurrences
+  if (certainty == FALSE){
+  certain <- occdf
+  } else {
+  certain <- occdf[(occdf[, certainty] == 1), ]
+  uncertain <- occdf[(occdf[, certainty] == 0), ]
+  }
+
+  #Determine y-axis label
+  if (units == "height") {
+    y_axis <- "Stratigraphic height (m)"
+  } else
+    y_axis <- "Bed number"
+
+  #Create plot
+  plot(x = NA, y = NA,
```

Comment:
I find it really hard to balance/design functions that make a plot for you. How much of the plotting functionality should you expose to the user? In this case, I would probably make "xlab" and "ylab" additional arguments of the new function (with ylab replacing the "units" argument). I would probably also add a "..." argument which you would then pass directly to `plot()`.